Design/WorkspacePropertyWithIndex.md

```diff
@@ -56,36 +62,66 @@ The points are taken from the original `IndexInfo` design document linked in the
 	-  `PropertyWidgetFactory` may need to handle `WorkspacePropertyWithIndex` specifically.
 	-  May need to introduce a concept of a group of property widgets.
 
-<sup>1</sup><sub>The distinction is made here between local (a subset of all data partitioned on an MPI Rank) and global (entire data set).</sub>
+<sup>1</sup><sub>The distinction is made here between local (a subset of all data partitioned on an MPI Rank) and global (entire data set). This will not be visible at the GUI or client API level.</sub>
 
 ### Example usage
 
 **Property Declaration**:
 
+Declaring the property in C++:
 ```cpp
-//Property name is fixed
-declareProperty(make_unique<WorkspacePropertyWithIndex>(IndexType::SpectrumNumber|IndexType::WorkspaceIndex));
+//Property name may be fixed?
+declareProperty(make_unique<WorkspacePropertyWithIndex<MatrixWorkspace>>("InputWorkspaceWithIndex", IndexType::SpectrumNumber|IndexType::WorkspaceIndex));
 ```  
 
 **Algorithm Dialog Box (GUI)**:
 
-**InputWorkspaceWithIndex** [ _________________ ]<br>
+**InputWorkspace** [ _________________ ]<br>
 ( **.** ) **SpectrumNumber** (  ) **WorkspaceIndex**<br>
 **Indices** [ _________________ ]
  
 **Client Code Access**:
 
-C++:
+*C++*:
+
+Setting the property manually:
+
+```cpp
+setProperty("InputWorkspaceWithIndex", ws, std::vector<SpectrumNumber>{1, 2, 3, 4});
+//or
+setProperty("InputWorkspaceWithIndex", ws, IndexType::SpectrumNumber, "1:33,42");
+//or
+setProperty("InputWorkspaceWithIndex", std::tie(ws, std::vector<SpectrumNumber>{1, 2, 3, 4}));
+//or 
+setProperty("InputWorkspaceWithIndex", std::tie(ws, IndexType::SpectrumNumber, "1:33,42"));
+```
+
+Accessing the property:
 
 ```cpp
 //C++11
 MatrixWorkspace_const_sptr inputWS;
 IndexSet indices;
-std::tie(inputWS, indices) = getProperty("InputWorkspaceWithIndex");
+std::tie(inputWS, indices) = std::pair<MatrixWorkspace_const_sptr, SpectrumIndexSet>(getProperty("InputWorkspaceWithIndex"));
```

Comment:
std::tie automatically evaluates to a tuple (although it can upack a pair). So if we use std::tuple in the implementation, I believe we can avoid writing the types twice.